packages/rich-text/src/plugins/Hyperlink/components/EntityHyperlink.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+
+import { FieldExtensionSDK, Link } from '@contentful/app-sdk';
+import { Tooltip, TextLink } from '@contentful/f36-components';
+import { useReadOnly } from 'slate-react';
+
+import { useContentfulEditor } from '../../../ContentfulEditorProvider';
+import { useSdkContext } from '../../../SdkProvider';
+import { CustomRenderElementProps } from '../../../types';
+import { addOrEditLink } from '../HyperlinkModal';
+import { useEntityInfo } from '../useEntityInfo';
+import { styles } from './styles';
+
+type HyperlinkElementProps = CustomRenderElementProps<{
+  uri?: string;
+  target?: Link;
+  onEntityFetchComplete?: VoidFunction;
+}>;
+
+export function EntityHyperlink(props: HyperlinkElementProps) {
+  const editor = useContentfulEditor();
+  const isReadOnly = useReadOnly();
+  const sdk: FieldExtensionSDK = useSdkContext();
+  const { target } = props.element.data;
+  const { onEntityFetchComplete } = props;
+
+  const tooltipContent = useEntityInfo({
+    target,
+    sdk,
+    onEntityFetchComplete,
+  });
+
+  if (!target) return null;
```

Comment:
Should we return `props.children` for the text to be rendered?

Reply:
* The check here is only to avoid TS complaining. It existed previously, I only moved logic around
* It's not possible to end up with a hyperlink node with no target based on our [schema rules](https://github.com/contentful/rich-text/blob/bca1072322e85ce9b3c9cc03d50e2ba232f252f8/packages/rich-text-types/src/nodeTypes.ts#L143-L157)
* If for whatever reason we did, then we also have to account for the possibility of not having `props.children at all and React won't like it when a component returns undefined.